<launch>

  <node pkg="view_asteroid"
    type="view_asteroid"
    name="view_asteroid"
    output="screen">

    <!-- Stereo baseline -->
    <rosparam param="cam_baseline"> 0.1 </rosparam>

  </node>

  <arg name="height" default="720" />
  <arg name="width" default="1280" />
  <arg name="f_length" default="960" />
  <arg name="half_height" default="$(eval 0.5*arg('height'))" />
  <arg name="half_width" default="$(eval 0.5*arg('width'))" />

  <group ns="camera1">
  <node name="camera_info" pkg="rostopic" type="rostopic"
      args="pub camera_info sensor_msgs/CameraInfo
     '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'camera2'},
      height: $(arg height), width: $(arg width), distortion_model: 'plumb_bob',
      D: [0],
      K: [$(arg f_length),             0.0, $(arg half_width), 
          0.0,             $(arg f_length), $(arg half_height),
          0.0,                         0.0,               1.0],
      R: [1.0, 0.0, 0.0,
          0.0, 1.0, 0.0,
          0.0, 0.0, 1.0],
      P: [$(arg f_length),             0.0,  $(arg half_width), 0.0,
          0.0,             $(arg f_length), $(arg half_height), 0.0, 
          0.0,                         0.0,                1.0, 0.0],
      binning_x: 0, binning_y: 0,
      roi: {x_offset: 0, 
            y_offset: 0, 
            height: $(arg height), 
            width: $(arg width), 
            do_rectify: true}}' -r 2"
      output="screen"/>
  </group>

  <group ns="camera2">
  <node name="camera_info" pkg="rostopic" type="rostopic"
      args="pub camera_info sensor_msgs/CameraInfo
     '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'camera2'},
      height: $(arg height), width: $(arg width), distortion_model: 'plumb_bob',
      D: [0],
      K: [$(arg f_length),             0.0, $(arg half_width), 
          0.0,             $(arg f_length), $(arg half_height),
          0.0,                         0.0,               1.0],
      R: [1.0, 0.0, 0.0,
          0.0, 1.0, 0.0,
          0.0, 0.0, 1.0],
      P: [$(arg f_length),             0.0,  $(arg half_width), 0.0,
          0.0,             $(arg f_length), $(arg half_height), 0.0, 
          0.0,                         0.0,                1.0, 0.0],
      binning_x: 0, binning_y: 0,
      roi: {x_offset: 0, 
            y_offset: 0, 
            height: $(arg height), 
            width: $(arg width), 
            do_rectify: true}}' -r 2"
      output="screen"/>
  </group>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find view_asteroid)/rviz/rviz_config.rviz"/>
</launch>

